components/gitpod-protocol/src/teams-projects-protocol.ts

```diff
@@ -15,6 +15,11 @@ export interface ProjectConfig {
 
 export interface ProjectSettings {
     enablePrebuilds?: boolean;
+    /**
+     * Wether prebuilds (if enabled) should only be started on the default branch.
+     * Defaults to `true` on project creation.
+     */
+    prebuildDefaultBranchOnly?: boolean;
```

Comment:
here (similar to enablkePrebuilds) we introduce an optional property where `undefined` means `true`. I think that is confusing. Can we make it `prebuildAllBranches?: boolean` or as we are going to allow a pattern already use a pattern internally and make it `*`?

Reply:
Alternatively, introduce a non-boolean property (i.e. the strategy thing below).

Reply:
I would like to remove the optional from `prebuildAllBranches`. This was a concern of rolling out to Gitpod Cloud. 

Once we agree on a data migration path, we can remove the `?`. 

I'm also OK with using constant to represent those states, e.g. `all-branches`, `default-branch`, `matched-branches`. Actually this would be even simpler, as we are using a `<select>` in frontend, which maps to a tuple of boolean to represents this state.